src/vs/workbench/parts/terminal/electron-browser/terminalProcess.js

```diff
@@ -48,6 +48,7 @@ ptyProcess.on('data', function (data) {
 });
 
 ptyProcess.on('exit', function (exitCode) {
+	ptyProcess.kill();
```

Comment:
I'll try this out on Windows, haven't tested it yet. What do you think would be a good timeout before we kill `terminalProcess`? 500 ms?